plugins/cilium-cni/cmd/cmd.go

```diff
@@ -428,170 +474,163 @@ func Add(args *skel.CmdArgs) (err error) {
 	}
 	defer netNs.Close()
 
-	if err = netns.RemoveIfFromNetNSIfExists(netNs, args.IfName); err != nil {
-		return fmt.Errorf("failed removing interface %q from namespace %q: %w",
-			args.IfName, args.Netns, err)
-	}
-
-	var ipam *models.IPAMResponse
-	var releaseIPsFunc func(context.Context)
-	if conf.IpamMode == ipamOption.IPAMDelegatedPlugin {
-		ipam, releaseIPsFunc, err = allocateIPsWithDelegatedPlugin(context.TODO(), conf, n, args.StdinData)
-	} else {
-		ipam, releaseIPsFunc, err = allocateIPsWithCiliumAgent(c, cniArgs, "")
+	res := &cniTypesV1.Result{}
+	config, err := cmd.cfg.GetConfiguration(ConfigurationParams{log, conf, args, cniArgs})
```

Comment:
nit: given this might return more than one endpoint configuration

```suggestion
	configs, err := cmd.cfg.GetConfiguration(ConfigurationParams{log, conf, args, cniArgs})
```